src/cascadia/TerminalSettingsEditor/Actions.xaml

```diff
@@ -181,44 +119,271 @@
+                        <TextBlock Grid.Column="0"
+                                   Style="{StaticResource KeyBindingNameTextBlockStyle}"
+                                   Text="{x:Bind Name, Mode=OneWay}" />
 
-            <!--
-                The Nav_OpenJSON resource just so happens to have a .Content
-                and .Tooltip that are _exactly_ what we're looking for here.
-            -->
+                        <!--  Key Chord Text  -->
+                        <Border Grid.Column="1"
+                                Padding="2,0,2,0"
+                                HorizontalAlignment="Right"
+                                VerticalAlignment="Center"
+                                Style="{ThemeResource KeyChordBorderStyle}"
+                                Visibility="{x:Bind IsInEditMode, Mode=OneWay, Converter={StaticResource InvertedBooleanToVisibilityConverter}}">
 
-            <HyperlinkButton x:Uid="Nav_OpenJSON"
-                             Click="_OpenSettingsClick" />
+                            <TextBlock FontSize="14"
+                                       Style="{ThemeResource KeyChordTextBlockStyle}"
+                                       Text="{x:Bind KeyChordText, Mode=OneWay}"
+                                       TextWrapping="WrapWholeWords" />
+                        </Border>
 
-            <!--  Keybindings  -->
+                        <!--  Edit Mode: Key Chord Text Box  -->
+                        <TextBox Grid.Column="1"
+                                 DataContext="{x:Bind Mode=OneWay}"
+                                 PreviewKeyDown="KeyChordEditor_PreviewKeyDown"
+                                 Style="{StaticResource KeyChordEditorStyle}"
+                                 Text="{x:Bind ProposedKeys, Mode=TwoWay}"
+                                 Visibility="{x:Bind IsInEditMode, Mode=OneWay}" />
 
-            <!--
-                NOTE: Globals_Keybindings.Header is not defined, because that
-                would result in the page having "Keybindings" displayed twice, which
-                looks quite redundant
-            -->
-            <ContentPresenter x:Uid="Globals_Keybindings"
-                              Margin="0">
-
-                <ListView HorizontalAlignment="Stretch"
-                          VerticalAlignment="Stretch"
-                          AllowDrop="False"
-                          CanReorderItems="False"
-                          IsItemClickEnabled="False"
-                          ItemTemplate="{StaticResource GeneralItemTemplate}"
-                          ItemsSource="{x:Bind FilteredActions, Mode=OneWay}"
-                          SelectionMode="None" />
-
-            </ContentPresenter>
+                        <!--  Edit Button  -->
+                        <Button x:Uid="Actions_EditButton"
+                                Grid.Column="2"
+                                AutomationProperties.Name="{x:Bind EditButtonName}"

```

Comment:
ugh this old chestnut

i  get it, but I hate it. This is *not* the responsibility of the viewmodel.